src/Tools/dotnet-monitor/IDiagnosticLifetimeService.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor
+{
+    /// <summary>
+    /// Callbacks for the lifetime events related to the diagnostic connection.
+    /// </summary>
+    public interface IDiagnosticLifetimeService
+    {
+        /// <summary>
+        /// Invoked just as the target process is discovered and is available for diagnostics.
+        /// </summary>
+        /// <remarks>
+        /// This is called just before the ResumeRuntime command is invoked.
+        /// </remarks>
+        ValueTask StartAsync(CancellationToken cancellationToken);
```

Comment:
is this a test only interface?

Reply:
No, this is a callback interface for scoped services to know when we've detected a diagnosable process and when to stop it's implementation due to the process shutting down/loss of connectivity/etc. It's similar to `IEndpointInfoSourceCallbacks`, which I hope will be superseded by this interface.